neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/dialect/impl/Neo4jAssociationQueries.java

```diff
@@ -96,17 +120,22 @@ private static String initRemoveAssociationQuery(EntityKeyMetadata ownerEntityKe
 		appendLabel( ownerEntityKeyMetadata, queryBuilder );
 		appendProperties( ownerEntityKeyMetadata, queryBuilder );
 		queryBuilder.append( ") - " );
-		queryBuilder.append( "[r" );
+		queryBuilder.append( "[a" );
 		queryBuilder.append( ":" );
 		appendRelationshipType( queryBuilder, associationKeyMetadata );
 		queryBuilder.append( "]" );
 		if ( associationKeyMetadata.getAssociationKind() == AssociationKind.EMBEDDED_COLLECTION ) {
 			queryBuilder.append( " - (e:" );
 			queryBuilder.append( EMBEDDED );
-			queryBuilder.append( ") DELETE r, e" );
+			queryBuilder.append( ")" );
+			queryBuilder.append( " WITH a,e" );
+			queryBuilder.append( "  MATCH path=(e) -[*0..]-> (:EMBEDDED) ");
+			queryBuilder.append( "  FOREACH ( r IN relationships(path) | DELETE r )" );
```

Comment:
Yes,
this part:
```
MATCH path=(e) -[*0..]-> (:EMBEDDED) 
```

`e` is already defined as an EMBEDDED in the previous part of the query, while the syntax `-[*0..]` select any path tha connect `e` to a (:EMBEDDED) with any number of relationships.